docs/contributor_guide/release.rst

```diff
@@ -0,0 +1,39 @@
+.. release_guide
+
+Release Process
+---------------
+
+This page outlines the process for creating a new Fairlearn release.
+The following steps must be performed on a clone of Fairlearn, not
+a fork.
+
+#. Create an annotated tag using the command at the branch point:
+
+    :code:`git tag -a v<x.y.z> -m "Branch point for v<x.y.z> release"`
```

Comment:
interesting, I'd assume you'd tag the branch, not `main`, as in first branch, test, then tag. Creating the tag has a few consequences for bots, and I think they should be created kind of as a last step.

Reply:
We can certainly do that instead. Tagging the branch point may just be leftover paranoia from my days working with CVS.